pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -198,6 +204,44 @@ public class Utils {
         r.addAction(new ExecutionModelAction(stages))
     }
 
+    static boolean withinAStage() {
+        CpsThread thread = CpsThread.current()
+        CpsFlowExecution execution = thread.execution
+
+        DepthFirstScanner scanner = new DepthFirstScanner();
```

Comment:
Or rather, I didn't understand what the differences between the two *mean* in this case.